lib/package.js

```diff
@@ -129,93 +132,266 @@ const async = require('async'),
+                createAppDir.bind(this),
+                checkELFHeader.bind(this),
+                fillAssetsField.bind(this),
+                copyAssets.bind(this),
+                copyApp.bind(this),
+                excludeIpkFileFromApp.bind(this),
+                createPackageDir.bind(this),
+                fillPackageDir.bind(this),
+                findServiceDir.bind(this, this.services),
+                loadServiceInfo.bind(this),
+                checkServiceInfo.bind(this),
+                createServiceDir.bind(this),
+                copyService.bind(this),
+                addServiceInPkgInfo.bind(this),
+                removeServiceFromAppDir.bind(this),
+                copyData.bind(this, inDirs, options.force),
+                loadPackageProperties.bind(this, inDirs),
+                excludeFromApp.bind(this),
+                outputPackage.bind(this, destination),
+                encryptPackage.bind(this),
+                copyOutputToDst.bind(this, destination),
+                recoverUmask.bind(this),
+                cleanupTmpDir.bind(this)
+            ], function(err) {
+                if (err) {
+                    // TODO: call cleanupTmpDir() before returning
+                    setImmediate(next, err);
+                    return;
+                }
+                // TODO: probably some more checkings are needed
+                setImmediate(next, null, {ipk: this.ipk, msg: "Success"});
+            }.bind(this));
+        },
+        analyzeIPK: function(options, next) {
+            log.info("package#Packager#analyzeIPK()");
+            const ipkConfigFile = path.resolve(__dirname, "../files/conf/ipk.json"),
+                ipkFile = options.info ? options.info : options.infodetail;
+            let ipkConfig, tmpDirPath;
+
+            async.series([
+                _checkFiles,
+                _checkTmpDir,
+                _unpackIpk,
+                _unpackTar,
+                _analyzeMetaFile,
+                _removeTmpDir,
+            ], function(err, results) {
+                log.silly("package#analyzeIPK()", "err:", err, ", results:", results);
+                if (err) {
+                    return next(err);
+                }
+                return next(null, {msg: results[4].trim()});
+            }.bind(this));
+
+            function _checkFiles(next) {
+                log.info("package#analyzeIPK()#_checkFiles()");
+                if (!fs.existsSync(ipkFile)) {
+                    return next(errHndl.getErrMsg("NOT_EXIST_PATH", ipkFile));
+                }
+                if (!fs.existsSync(ipkConfigFile)) {
+                    return next(errHndl.getErrMsg("NOT_EXIST_PATH", ipkConfigFile));
+                }
+
+                ipkConfig = JSON.parse(fs.readFileSync(ipkConfigFile));
+                tmpDirPath = temp.path(ipkConfig.tmpPath);
+                log.verbose("package#analyzeIPK()#_checkFiles()", "tmpDirPath: ", tmpDirPath);
```

Comment:
Could you move the lines to in _checkTmpDir ?

Reply:
#253 line keep in the function. 
#254 line is removed and add more information in _checkTmpDir log.